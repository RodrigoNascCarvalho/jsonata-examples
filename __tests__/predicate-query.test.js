const { testJsonataExpr } = require('./jsonata-helper')
const inputJson = require('example-json')

describe('Predicate Queries Examples', () => {
    test('PQ01: Select all Products with the ProductID for Bowler Hat', () => {
        testJsonataExpr(inputJson, 'Account.Order.Product[ProductID=858383]')
    })
    
    test('PQ02: Select all Orders regardless of JSON structure', () => {
        testJsonataExpr(inputJson, '*.Order')
    })
    
    test('PQ03: Select all properties from Orders regardless of properties names', () => {
        testJsonataExpr(inputJson, 'Account.Order.*')
    })
    
    test('PQ04: Select all Products regardless of structure level', () => {
        testJsonataExpr(inputJson, '**.Product')
    })
})
