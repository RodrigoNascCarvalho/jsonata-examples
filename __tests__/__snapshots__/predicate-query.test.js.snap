// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Predicate Queries Examples PQ01: Select all Products with the ProductID for Bowler Hat 1`] = `
Array [
  Object {
    "Description": Object {
      "Colour": "Purple",
      "Depth": 210,
      "Height": 200,
      "Weight": 0.75,
      "Width": 300,
    },
    "Price": 34.45,
    "Product Name": "Bowler Hat",
    "ProductID": 858383,
    "Quantity": 2,
    "SKU": "0406654608",
  },
  Object {
    "Description": Object {
      "Colour": "Blue",
      "Depth": 210,
      "Height": 200,
      "Weight": 0.75,
      "Width": 300,
    },
    "Price": 34.45,
    "Product Name": "Bowler Hat",
    "ProductID": 858383,
    "Quantity": 4,
    "SKU": "040657863",
  },
  Object {
    "Description": Object {
      "Colour": "Yellow",
      "Depth": 210,
      "Height": 200,
      "Weight": 0.75,
      "Width": 300,
    },
    "Price": 34.45,
    "Product Name": "Bowler Hat",
    "ProductID": 858383,
    "Quantity": 2,
    "SKU": "0406654608",
  },
]
`;

exports[`Predicate Queries Examples PQ02: Select all Orders regardless of JSON structure 1`] = `
Array [
  Object {
    "OrderID": "order103",
    "Product": Array [
      Object {
        "Description": Object {
          "Colour": "Purple",
          "Depth": 210,
          "Height": 200,
          "Weight": 0.75,
          "Width": 300,
        },
        "Price": 34.45,
        "Product Name": "Bowler Hat",
        "ProductID": 858383,
        "Quantity": 2,
        "SKU": "0406654608",
      },
      Object {
        "Description": Object {
          "Colour": "Orange",
          "Depth": 210,
          "Height": 200,
          "Weight": 0.6,
          "Width": 300,
        },
        "Price": 21.67,
        "Product Name": "Trilby hat",
        "ProductID": 858236,
        "Quantity": 1,
        "SKU": "0406634348",
      },
    ],
  },
  Object {
    "OrderID": "order104",
    "Product": Array [
      Object {
        "Description": Object {
          "Colour": "Blue",
          "Depth": 210,
          "Height": 200,
          "Weight": 0.75,
          "Width": 300,
        },
        "Price": 34.45,
        "Product Name": "Bowler Hat",
        "ProductID": 858383,
        "Quantity": 4,
        "SKU": "040657863",
      },
      Object {
        "Description": Object {
          "Colour": "Black",
          "Depth": 210,
          "Height": 20,
          "Weight": 2,
          "Width": 30,
        },
        "Price": 107.99,
        "Product Name": "Cloak",
        "ProductID": 345664,
        "Quantity": 1,
        "SKU": "0406654603",
      },
    ],
  },
  Object {
    "OrderID": "order105",
    "Product": Array [
      Object {
        "Description": Object {
          "Colour": "Yellow",
          "Depth": 210,
          "Height": 200,
          "Weight": 0.75,
          "Width": 300,
        },
        "Price": 34.45,
        "Product Name": "Bowler Hat",
        "ProductID": 858383,
        "Quantity": 2,
        "SKU": "0406654608",
      },
      Object {
        "Description": Object {
          "Colour": "Green",
          "Depth": 210,
          "Height": 100,
          "Weight": 2,
          "Width": 30,
        },
        "Price": 207.99,
        "Product Name": "Cloak Dagger",
        "ProductID": 345666,
        "Quantity": 1,
        "SKU": "0406654604",
      },
    ],
  },
]
`;

exports[`Predicate Queries Examples PQ03: Select all properties from Orders regardless of properties names 1`] = `
Array [
  "order103",
  Object {
    "Description": Object {
      "Colour": "Purple",
      "Depth": 210,
      "Height": 200,
      "Weight": 0.75,
      "Width": 300,
    },
    "Price": 34.45,
    "Product Name": "Bowler Hat",
    "ProductID": 858383,
    "Quantity": 2,
    "SKU": "0406654608",
  },
  Object {
    "Description": Object {
      "Colour": "Orange",
      "Depth": 210,
      "Height": 200,
      "Weight": 0.6,
      "Width": 300,
    },
    "Price": 21.67,
    "Product Name": "Trilby hat",
    "ProductID": 858236,
    "Quantity": 1,
    "SKU": "0406634348",
  },
  "order104",
  Object {
    "Description": Object {
      "Colour": "Blue",
      "Depth": 210,
      "Height": 200,
      "Weight": 0.75,
      "Width": 300,
    },
    "Price": 34.45,
    "Product Name": "Bowler Hat",
    "ProductID": 858383,
    "Quantity": 4,
    "SKU": "040657863",
  },
  Object {
    "Description": Object {
      "Colour": "Black",
      "Depth": 210,
      "Height": 20,
      "Weight": 2,
      "Width": 30,
    },
    "Price": 107.99,
    "Product Name": "Cloak",
    "ProductID": 345664,
    "Quantity": 1,
    "SKU": "0406654603",
  },
  "order105",
  Object {
    "Description": Object {
      "Colour": "Yellow",
      "Depth": 210,
      "Height": 200,
      "Weight": 0.75,
      "Width": 300,
    },
    "Price": 34.45,
    "Product Name": "Bowler Hat",
    "ProductID": 858383,
    "Quantity": 2,
    "SKU": "0406654608",
  },
  Object {
    "Description": Object {
      "Colour": "Green",
      "Depth": 210,
      "Height": 100,
      "Weight": 2,
      "Width": 30,
    },
    "Price": 207.99,
    "Product Name": "Cloak Dagger",
    "ProductID": 345666,
    "Quantity": 1,
    "SKU": "0406654604",
  },
]
`;

exports[`Predicate Queries Examples PQ04: Select all Products regardless of structure level 1`] = `
Array [
  Object {
    "Description": Object {
      "Colour": "Purple",
      "Depth": 210,
      "Height": 200,
      "Weight": 0.75,
      "Width": 300,
    },
    "Price": 34.45,
    "Product Name": "Bowler Hat",
    "ProductID": 858383,
    "Quantity": 2,
    "SKU": "0406654608",
  },
  Object {
    "Description": Object {
      "Colour": "Orange",
      "Depth": 210,
      "Height": 200,
      "Weight": 0.6,
      "Width": 300,
    },
    "Price": 21.67,
    "Product Name": "Trilby hat",
    "ProductID": 858236,
    "Quantity": 1,
    "SKU": "0406634348",
  },
  Object {
    "Description": Object {
      "Colour": "Blue",
      "Depth": 210,
      "Height": 200,
      "Weight": 0.75,
      "Width": 300,
    },
    "Price": 34.45,
    "Product Name": "Bowler Hat",
    "ProductID": 858383,
    "Quantity": 4,
    "SKU": "040657863",
  },
  Object {
    "Description": Object {
      "Colour": "Black",
      "Depth": 210,
      "Height": 20,
      "Weight": 2,
      "Width": 30,
    },
    "Price": 107.99,
    "Product Name": "Cloak",
    "ProductID": 345664,
    "Quantity": 1,
    "SKU": "0406654603",
  },
  Object {
    "Description": Object {
      "Colour": "Yellow",
      "Depth": 210,
      "Height": 200,
      "Weight": 0.75,
      "Width": 300,
    },
    "Price": 34.45,
    "Product Name": "Bowler Hat",
    "ProductID": 858383,
    "Quantity": 2,
    "SKU": "0406654608",
  },
  Object {
    "Description": Object {
      "Colour": "Green",
      "Depth": 210,
      "Height": 100,
      "Weight": 2,
      "Width": 30,
    },
    "Price": 207.99,
    "Product Name": "Cloak Dagger",
    "ProductID": 345666,
    "Quantity": 1,
    "SKU": "0406654604",
  },
]
`;
